src/plugins/mf_priority.cpp

```diff
@@ -198,12 +198,181 @@ int check_queue_factor (flux_plugin_t *p,
 }
 
 
+/*
+ * Add held job IDs to a JSON array to be added to a bank_info JSON object.
+ */
+static json_t *add_held_jobs (
+                            const std::pair<std::string, struct bank_info> &b)
+{
+    json_t *held_jobs = NULL;
+
+    // add any held jobs to a JSON array
+    if (!(held_jobs = json_array ()))
+        goto error;
+
+    for (auto const &j : b.second.held_jobs) {
+        json_t *jobid = json_integer (j);
+
+        if (!jobid)
+            goto error;
+
+        if (json_array_append (held_jobs, jobid) < 0) {
+            json_decref (jobid);
+            goto error;
+        }
+    }
+
+    return held_jobs;
+error:
+    json_decref (held_jobs);
+    return NULL;
+}
+
+
+/*
+ * Create a JSON object for a bank that a user belongs to.
+ */
+static json_t *pack_bank_info_object (
+                            const std::pair<std::string, struct bank_info> &b)
+{
+    json_t *bank_info, *held_jobs = NULL;
+
+    bank_info = json_object ();
+    if (!bank_info)
+        goto error;
+
+    held_jobs = add_held_jobs (b);
+    if (held_jobs == NULL)
+        goto error;
+
+    if (!(bank_info = json_pack ("{s:s, s:f, s:i, s:i, s:i, s:i, s:o, s:i}",
+                                 "bank", b.first.c_str (),
+                                 "fairshare", b.second.fairshare,
+                                 "max_run_jobs", b.second.max_run_jobs,
+                                 "cur_run_jobs", b.second.cur_run_jobs,
+                                 "max_active_jobs", b.second.max_active_jobs,
+                                 "cur_active_jobs", b.second.cur_active_jobs,
+                                 "held_jobs", held_jobs,
+                                 "active", b.second.active))) {
+            goto error;
+    }
+
+    return bank_info;
+error:
+    json_decref (held_jobs);
+    json_decref (bank_info);
+    return NULL;
+}
+
+
+/*
+ * For each bank that a user belongs to, create a JSON object for each bank and
+ * add it to the user JSON object.
+ */
+static json_t *banks_to_json (
+                    flux_plugin_t *p,
+                    std::pair<int, std::map<std::string, struct bank_info>> &u)
+{
+    json_t *bank_info, *banks = NULL;
+
+    banks = json_array (); // array of banks that user belongs to
+    if (!banks)
+        goto error;
+
+    for (auto const &b : u.second) {
+        if (!(bank_info = json_object ())) // JSON object for one bank
+            goto error;
```

Comment:
Like the other function, you _could_ declare `bank_info` inside the loop, then use `json_array_append_new()` as described earlier.